### PR TITLE
Improve logs and refactor (open-backport-pull-request)

### PR DESCRIPTION
**Pull request by int128**
## Problem to solve
open-backport-pull-request action does not show any log when there is no diff between base branch and head branch.

## How to solve
Show more logs.

### CODE REVIEW

File: open-backport-pull-request/src/run.ts [hunk 1]
```diff
@@ -22,37 +20,30 @@ type Outputs = {
...
-  context: Context
-  octokit: Octokit
-}
-
-const hasDiff = async (params: hasDiffParams): Promise<boolean> => {
-  const compare = await params.octokit.request('GET /repos/{owner}/{repo}/compare/{basehead}', {
-    owner: context.repo.owner,
-    repo: context.repo.repo,
-    basehead: `${params.baseBranch}...${params.headBranch}`,
+  core.info(`Comparing ${inputs.baseBranch} and ${inputs.headBranch} branch`)
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:25]: Added this log.

File: open-backport-pull-request/src/run.ts [hunk 2]
```diff
@@ -22,37 +20,30 @@ type Outputs = {
...
+  const { data: compare } = await octokit.rest.repos.compareCommitsWithBasehead({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    basehead: `${inputs.baseBranch}...${inputs.headBranch}`,
   })
-  return !!(compare.data.files && compare.data.files.length > 0)
+  if (compare.files === undefined || compare.files.length === 0) {
+    core.info(`No changes between ${inputs.baseBranch} and ${inputs.headBranch} branch. Do nothing.`)
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:32]: Added this log.

File: open-backport-pull-request/src/run.ts [hunk 3]
```diff
@@ -22,37 +20,30 @@ type Outputs = {
...
-        baseBranch,
-        skipCI: inputs.skipCI,
-        mergePullRequest: inputs.mergePullRequest,
-        context,
-      },
-      octokit,
-    )
-  }
-}
+  const octokit = github.getOctokit(inputs.githubToken, {}, octokitPluginRetry.retry)
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:23]: It needs to pass a plugin to third argument.

```ts
export function getOctokit(
  token: string,
  options?: OctokitOptions,
  ...additionalPlugins: OctokitPlugin[]
): InstanceType<typeof GitHub> {
```

https://github.com/actions/toolkit/blob/ae38557bb0dba824cdda26ce787bd6b66cf07a83/packages/github/src/github.ts#L15-L19

File: open-backport-pull-request/src/run.ts [hunk 4]
```diff
@@ -22,37 +20,30 @@ type Outputs = {
 }
 
 export const run = async (inputs: Inputs): Promise<Outputs | undefined> => {
-  const octokit = getOctokit(inputs.githubToken, octokitPluginRetry.retry)
-  const { baseBranch, headBranch } = inputs
-
-  if (await hasDiff({ headBranch, baseBranch, octokit, context })) {
-    return await openPullRequest(
```
Comment (int128) [inline open-backport-pull-request/src/run.ts:29]: Moved to line 37 https://github.com/quipper/monorepo-deploy-actions/pull/1435/files#diff-f2944a3a81132688fe8339ed478459a84154c64c8c2753c2f96f0a74a067df83R37

**Review by panicboat — APPROVED** [review]

**Pull request merged**